README.md
# Web Application

Coinage: A PHP driven application that allows users to store unique phrases in a database.
Find the application on [PHP Fog] (http://coinage.phpfogapp.com/index.php)

#Author Information

Josh Davis

Contact: [davis.webinfo@gmail.com] (davis.webinfo@gmail.com)

Website:  [JJAD] (http://jjad.ca)

#Source Code

[Github] (https://github.com/JoshDavis/web-app)

#License

[License] (https://github.com/JoshDavis/web-app/blob/master/license.txt)


changelog.md
#Change Log

Version 1.0.0



add.php
<?php

require_once 'includes/db.php';

$sql = $db->query('
	SELECT id, word, definition
	FROM coinage
	ORDER BY id DESC 


');

$results = $sql->fetchAll();


?><!DOCTYPE HTML>
<html>
<head>
    <meta charset="utf-8">
    <title></title>
    <link href="css/general.css" rel="stylesheet">
    <link href='http://fonts.googleapis.com/css?family=Bevan' rel='stylesheet' type='text/css'>
    <link href='http://fonts.googleapis.com/css?family=Nobile:400,700' rel='stylesheet' type='text/css'>
    <script src="js/modernizr-2.5.3.js"></script>
</head>

<body>
<a name="top"></a>
	<header>
        <h1><a href="index.php">Coinage</a></h1>
        
        <p class="definition">a. A new word or phrase.
        </br>b. The invention of new words.</p><p class="subtitle">(koi 'nij)</p> 
    </header>
    
    <section class="entries">
    <div class="wrapper-data">
   
	<?php foreach ($results as $coinage) : ?>
	<article>
        <h3>Word/Phrase: </h3>
        <i class="word-data"><?php echo $coinage['word']; ?></a></i>
            
        <h3>Definiton: </h3>
        <i class="def-data"><?php echo $coinage['definition'];?></i>    
    </article> 
	<?php endforeach; ?>
	</div>
    
    <aside>
    	<p>Browse the list and then add something of your own. Be creative.</p>
        <div id="add-page-button-two"><a href="index.php">I'd like to add a new word or phrase[...]</a></div>
        
        <p><small>Try and keep it clean, people. Or don't, what do I care?</small></p>
        <a href="https://twitter.com/joshuajadavis" class="twitter-follow-button" data-show-count="false" data-size="large">Follow @joshuajadavis</a>
		<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0];if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src="//platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>
	
    <div id="add-page-button-two">
    	<a href="http://github.com/JoshDavis">Find this app on Github.</a>
    </div>
		<p><small>Open Source &middot; Josh Davis &middot; 2012</small></p>
    </aside>
    <footer>
    	<a href="#top"><img src="images/back.png" alt="back to top"></a>
    </footer>
    </section>
    
    

</body>
</html>

single.php
<?php

require_once 'includes/db.php';

$id = filter_input(INPUT_GET, 'id', FILTER_SANITIZE_NUMBER_INT);


//Prepare allows us to have variables in out SQL
//We can create placeholders and later fill them with some content.
//Prepare is used to protect against SQL injection attacks.
$sql = $db->prepare('
SELECT id, word, definition
FROM coinage
WHERE id = :id
');

$sql ->bindValue(':id', $id, PDO::PARAM_INT);
$sql -> execute();
$results = $sql->fetch();

?>




<!DOCTYPE HTML>
<html>
<head>
    <meta charset="utf-8">
    <title><?php echo $results ['word'];?>&middot; in Words/Phrases</title>
    <link href="css/general.css" rel="stylesheet">
</head>

<body>

	<h1><?php echo $results['word']; ?></h1>
    <dl>
    	    <dt>Word/Phrase</dt>
            <dd><?php echo $results['word'];?></dd>
            
            <dt>Definition</dt>
            <dd><?php echo $results['definition'];?></dd>
    </dl>

    <center>
    <a href="add.php">Go back to inventory...</a>
	</center>
	


</body>
</html>

index.php
<?php

$word = filter_input(INPUT_POST, "word", FILTER_SANITIZE_STRING);
$definition = filter_input(INPUT_POST, "definition", FILTER_SANITIZE_STRING);
$errors = array();

require_once 'includes/db.php';

if ($_SERVER['REQUEST_METHOD']== 'POST'){
	 if (strlen($word) < 1 || strlen ($word) > 256) {
		 $errors['word'] = true;
	 }
	 if (strlen($definition) < 1 || strlen ($definition) > 256) {
		 $errors['definition'] = true;
	 }
	 if (empty($errors)) {
		require_once 'includes/db.php';
		
		$sql = $db->prepare('
		
			INSERT INTO coinage (word, definition)
			VALUES (:word, :definition)
		
		');
		$sql->bindValue(':word', $word, PDO::PARAM_STR);
		$sql->bindValue(':definition', $definition, PDO::PARAM_STR);
		
		$sql->execute();
		
		header('Location:add.php');
		exit;
	 }
}

?><!DOCTYPE HTML>
<html>
<head>
    <meta charset="utf-8">
    <title>Coinage &middot; Be Creative</title>
    <link href="css/general.css" rel="stylesheet">
    <link href='http://fonts.googleapis.com/css?family=Bevan' rel='stylesheet' type='text/css'>
    <link href='http://fonts.googleapis.com/css?family=Nobile:400,700' rel='stylesheet' type='text/css'>
    <script src="js/modernizr-2.5.3.js"></script>
</head>

<body>
	<header>
        <h1>Coinage</h1>
        
        <p class="definition">a. A new word or phrase.
        </br>b. The invention of new words.</p><p class="subtitle">(koi 'nij)</p>
    </header>
    
    <div class="container">
        <form method="post" action="index.php">
    
        <div>
            <label for="word">New word or phrase:</label>
            <?php if(isset($errors['word'])) : ?><strong class="error">Is Required.</strong><?php endif;?>
            <input id="word" name="word" required value="<?php echo $word; ?>">
        </div>
        <div>
            <label for="definition">Definition, please:</label></br> 
            <?php if(isset($errors['definition'])) : ?><strong class="error">Is Required.</strong><?php endif;?>
            <input id="definition" name="definition" required value="<?php echo $definition; ?>">
         </div>
        
        <div class="button-holder">
            <p>Let's make it famous:</p>
            <button type="submit">Add to Inventory</button>
        </div>
        
        <p class="check"><a href="add.php">Check inventory of words and phrases.</a></p>
        </form>
        </div>
        
        <div class="about-me">
            <h4><a href="http://jjad.ca/about.php"><img src="images/about-rocket.png" alt=""></a></h4>
   		</div>
    </div>
    
</body>
</html>